data_source.go

```diff
@@ -477,20 +490,20 @@ func (ds *AnySource) PrepareRun() error {
 
 // Stop ends the data supply.
 func (ds *AnySource) Stop() error {
-	if ds.Running() {
-		close(ds.abortSelf)
+	if !ds.Running() {
+		return fmt.Errorf("Anysource not running, cannot stop")
 	}
-	ds.runDone.Wait()
```

Comment:
Why isn't the new code calling `ds.Wait()` ?

Reply:
I'm not sure what `ds.Wait()` is, I couldn't find the definition, but the tests panic when I use it.

Reply:
It's a `sync.WaitGroup`. It lets something wait on multiple events to happen, potentially in separate goroutines. Not sure whether we need it here.

Reply:
`runeDone` is the `sync.WaitGroup` and is what I'm waiting on here. I think `ds.Wait()` is actually the function I'm defining here, and I was causing a stack overflow to get those panics.